### DISCUSSION TITLE
backup files should be always copied

### DISCUSSION THREAD

**Discussion by aster-void** [helix-editor/helix#12047] [General] [answered]
see helix-view/document.rs :959 for the code

current implementation of moving it is causing too many issues:
- vite watch fails to detect changes
- inotifywatch fails to detect writes
- file perms change on save / see #11988 
- hard link breaks (even though it's explicitly checked??)

and it doesn't serve much purpose because:
- doesn't save disk space because the file is overwritten instantly
- we don't edit gigabytes of file very often

if the speed is critical on large files, move it only when the file is that large.

**Comment by kirawi** [comment]
Writing to backup files is what Vim/Neovim does as well as many other editors. The issue is simply that there are some edge-cases that aren't covered by the current algorithm. File watchers should be designed with the ability to detect such changes from my understanding, so the onus is on them to address that, not the editor.

> hard link breaks (even though it's explicitly checked??)

You need to use master for the fix.

Most issues with the system will be fixed with a planned rewrite I have for it that will more closely match Vim's backup system.

**Reply by aster-void** [reply]
I'm not against creating backup files, it's just that it should be always copied, not moved

**Comment by kirawi** [comment, marked as answer]
See https://github.com/helix-editor/helix/pull/11374
